packages/next/lib/resolve-request.ts

```diff
@@ -8,6 +8,17 @@ export function resolveRequest(req: string, issuer: string): string {
   if (process.versions.pnp) {
     const { resolveRequest: pnpResolveRequest } = require(`pnpapi`)
     return pnpResolveRequest(req, issuer, { considerBuiltins: false })
+  } else if (process?.env?.npm_config_user_agent?.startsWith('pnpm/')) {
```

Comment:
Is this condition true only when running Next.js via `pnpm run <script>`? My concern is that this leads to false negatives: `node_modules` could be installed via pnpm, but we could be running Next.js via some other task runner, or directly by running `node_modules/.bin/next`.

Reply:
Yes, that would be a problem, it only fixes when running scripts via pnpm.  I would prefer your PR #16369 and what I mentioned above.    No conditional logic at all for pnpm.